src/invariant_service.rs

```diff
@@ -881,6 +884,50 @@ where
         }
     }
 
+    pub fn get_tickmap(&self, pool_key: PoolKey) -> Vec<(u16, u64)> {
+        let tick_spacing = pool_key.fee_tier.tick_spacing;
+
+        let max_chunk_index = get_max_chunk(tick_spacing);
+        let min_chunk_index = get_min_chunk(tick_spacing);
+
+        InvariantStorage::as_ref().tickmap_slice(min_chunk_index..=max_chunk_index, pool_key)
+    }
+
+    pub fn get_liquidity_ticks(&self, pool_key: PoolKey) -> Vec<LiquidityTick> {
```

Comment:
Is there such a large amount of gas here to return all Liquidity Ticks even in the pessimistic case, or is this simply the first iteration to test simulateSwap?